File: poi-backend/src/main/kotlin/org/mrmeowcat/poibackend/domain/service/SignUpService.kt
package org.mrmeowcat.poibackend.domain.service

import org.mrmeowcat.poibackend.application.dto.request.SignUpRequest
import org.mrmeowcat.poibackend.application.dto.response.SignUpValidationResponse

/**
 * Service for sign up handling.
 */
interface SignUpService {

    fun signUp(signUpRequest: SignUpRequest?) : SignUpValidationResponse

}

File: poi-webface/src/app/services/sign-up.service.ts
import { Injectable } from '@angular/core';
import { HttpService } from "./http.service";
import { SignUpRequest } from "../models/sign-up-request";
import { Urls } from "../utils/urls";

@Injectable()
export class SignUpService {

  constructor(private http: HttpService) {
  }

  signUp(signUpRequest: SignUpRequest): Promise<any> {
    return this.http.post(Urls.SIGN_UP_URL, signUpRequest).toPromise();
  }
}
